bindinfo/handle.go

```diff
@@ -217,6 +271,36 @@ func (h *BindHandle) appendBindMeta(hash string, meta *bindMeta) {
 	newCache[hash] = append(newCache[hash], meta)
 }
 
+// removeBindMeta removes the bindMeta from the cache.
+func (h *BindHandle) removeBindMeta(hash string, meta *bindMeta) {
+	h.bindInfo.Lock()
+	newCache := h.bindInfo.Value.Load().(cache).copy()
+	defer func() {
+		h.bindInfo.Value.Store(newCache)
+		h.bindInfo.Unlock()
+	}()
+
+	newCache.removeDeletedBindMeta(hash, meta)
+}
+
+// removeDeletedBindMeta removes all the bindMeta which originSQL and db are the same with the parameter's meta.
+func (c cache) removeDeletedBindMeta(hash string, meta *bindMeta) {
+	metas, ok := c[hash]
+	if !ok {
+		return
+	}
+
+	for i := len(metas) - 1; i >= 0; i-- {
+		if meta.isSame(meta) {
+			metas = append(metas[:i], metas[i+1:]...)
```

Comment:
If the slice reallocates, the `metas` will be different from `c[hash]`? And so with the `removeStaleBindMetas`.

Reply:
Never mind, remove won't reallocate.